### PR TITLE
Improve docs for `jnp.logspace` and `jnp.geomspace`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -5896,10 +5896,66 @@ def _linspace(start: ArrayLike, stop: ArrayLike, num: int = 50,
...
+
+  Args:
+    start: scalar or array. Used to specify the start value. The start value is
+      ``base ** start``.
+    stop: scalar or array. Used to specify the stop value. The end value is
+      ``base ** stop``.
+    num: int, optional, default=50. Number of values to generate.
+    endpoint: bool, optional, default=True. If True, then include the ``stop`` value
+      in the result. If False, then exclude the ``stop`` value.
+    base: scalar or array, optional. Specifies the base of the logarithm.
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:5914]: Mention that the default is 10

File: jax/_src/numpy/lax_numpy.py [hunk 2]
```diff
@@ -5896,10 +5896,66 @@ def _linspace(start: ArrayLike, stop: ArrayLike, num: int = 50,
...
+    An array of logarithm.
+
+  See also:
+    - :func:`jax.numpy.arange`: Generate ``N`` evenly-spaced values given a starting
+      point and a step value.
+    - :func:`jax.numpy.linspace`: Generate evenly-spaced values.
+    - :func:`jax.numpy.geomspace`: Generate geometrically-spaced values.
+
+  Examples:
+    List 5 logarithmically spaced values between 1 and 100:
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:5928]:
```suggestion
    List 5 logarithmically spaced values between 1 (``10 ** 0``) and 100 (``10 ** 2``):
```

File: jax/_src/numpy/lax_numpy.py [hunk 3]
```diff
@@ -5896,10 +5896,66 @@ def _linspace(start: ArrayLike, stop: ArrayLike, num: int = 50,
...
+    ...   jnp.logspace(0, 2, 5)
+    Array([  1.   ,   3.162,  10.   ,  31.623, 100.   ], dtype=float32)
+
+    List 5 logarithmically-spaced values between 1 and 100, excluding endpoint:
+
+    >>> with jnp.printoptions(precision=3, suppress=True):
+    ...   jnp.logspace(0, 2, 5, endpoint=False)
+    Array([ 1.   ,  2.512,  6.31 , 15.849, 39.811], dtype=float32)
+
+    List 7 logarithmically-spaced values between 1 and 100 with base 2:
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:5940]:
```suggestion
    List 7 logarithmically-spaced values between 1 (``2 ** 0``) and 4 (``2 ** 2``)
    by setting ``base=2``:
```

File: jax/_src/numpy/lax_numpy.py [hunk 4]
```diff
@@ -5922,9 +5978,52 @@ def _logspace(start: ArrayLike, stop: ArrayLike, num: int = 50,
   return lax.convert_element_type(ufuncs.power(base, lin), dtype)
 
 
-@util.implements(np.geomspace)
 def geomspace(start: ArrayLike, stop: ArrayLike, num: int = 50, endpoint: bool = True,
               dtype: DTypeLike | None = None, axis: int = 0) -> Array:
+  """Generate geometrically-spaced values.
+
+  JAX implementation of :func:`numpy.geospace`.
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:5985]:
```suggestion
  JAX implementation of :func:`numpy.geomspace`.
```

**Review by jakevdp — CHANGES_REQUESTED** [review]
Looks good!

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Done

File: jax/_src/numpy/lax_numpy.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Thanks! Done

File: jax/_src/numpy/lax_numpy.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Done

File: jax/_src/numpy/lax_numpy.py (hunk 4 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Thanks! Done

**Review by jakevdp — APPROVED** [review]

**Comment by rajasekharporeddy** [comment]
Hi @jakevdp, Fixed a doctest error and squashed the commits into one. Could you please trigger the tests once again?

**Pull request merged**